src/flashbar/index.tsx

```diff
@@ -1,85 +1,178 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 import clsx from 'clsx';
-import React from 'react';
-import { useMergeRefs } from '../internal/hooks/use-merge-refs';
-import { getBaseProps } from '../internal/base-component';
-import styles from './styles.css.js';
-import { Flash } from './flash';
-import { useContainerBreakpoints } from '../internal/hooks/container-queries';
+import React, { useEffect, useState } from 'react';
 import { applyDisplayName } from '../internal/utils/apply-display-name';
-import VisualContext from '../internal/components/visual-context';
-import useBaseComponent from '../internal/hooks/use-base-component';
-import { useReducedMotion, useVisualRefresh } from '../internal/hooks/use-visual-mode';
+import customCssProps from '../internal/generated/custom-css-properties';
+import { Flash } from './flash';
 import { FlashbarProps } from './interfaces';
+import { getBaseProps } from '../internal/base-component';
+import InternalIcon from '../icon/internal';
+import { TIMEOUT_FOR_ENTERING_ANIMATION } from './constant';
 import { TransitionGroup } from 'react-transition-group';
 import { Transition } from '../internal/components/transition';
-import { TIMEOUT_FOR_ENTERING_ANIMATION } from './constant';
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { useContainerBreakpoints } from '../internal/hooks/container-queries';
+import useFocusVisible from '../internal/hooks/focus-visible';
+import { useMergeRefs } from '../internal/hooks/use-merge-refs';
+import { useReducedMotion, useVisualRefresh } from '../internal/hooks/use-visual-mode';
+import { getVisualContextClassname } from '../internal/components/visual-context';
+import styles from './styles.css.js';
 
 export { FlashbarProps };
 
 export default function Flashbar({ items, ...restProps }: FlashbarProps) {
   const { __internalRootRef } = useBaseComponent('Flashbar');
   const [breakpoint, ref] = useContainerBreakpoints(['xs']);
-  const isRefresh = useVisualRefresh();
   const baseProps = getBaseProps(restProps);
-
   const mergedRef = useMergeRefs(ref, __internalRootRef);
+  const isFocusVisible = useFocusVisible();
+  const isVisualRefresh = useVisualRefresh();
+
+  /**
+   * All the flash items should have ids so we can identify which DOM element is being
+   * removed from the DOM to animate it. Motion will be disabled if any of the provided
+   * flash messages does not contain an `id`.
+   */
+  const motionDisabled =
+    useReducedMotion(ref as any) || !isVisualRefresh || (items && !items.every(item => 'id' in item));
+
+  /**
+   * The `enableStackingOption` property is a hidden boolean that allows for teams
+   * to beta test the flashbar stacking feature.
+   */
+  const enableStackingOption = (restProps as any).enableStackingOption;
```

Comment:
Even if it is a beta, better to come up with a solid future-proof name for this property. I think, it should be `stackItems`, but open for other suggestions too.

Reply:
An important distinction that this property only turns on the option to stack for users, but the actual piece of state that controls whether or not the notifications are stacked is separate. I know @connorlanigan had some feedback on the name as well. I will start a Slack thread to see if we can achieve a quick consensus.